include/MVC/SugarApplication.php

```diff
@@ -704,49 +706,77 @@ public static function getSuccessMessages() {
-        
-        self::validateMessageType($type);
+     * Set a message which reflects the status of the performed operation.
+     *
+     * @param string $message the message to be displayed to the user.
+     * For consistency with other messages, it should begin with a capital letter and end with a period.
+     *
+     * @param string $type the type of the message. Values allowed: error, info, alert, okay, warning.
+     *
+     * @param boolean $repeat if this is FALSE and the message is already set, then the message won't be repeated.
+     *
+    */
+    public static function appendMessage($message, $type = 'info', $repeat = true)
+    {
 
-        if (empty($_SESSION[$type]) || !is_array($_SESSION[$type])) {
-            $_SESSION[$type] = array();
-        }
-        if(!in_array($message, $_SESSION[$type])) {
-            $_SESSION[$type][] = $message;
+        if (!empty($message)) {
+            $type = preg_replace('[^a-z]', '', strtolower($type));
+
+            if (!isset($_SESSION['suite_messages'][$type])
+                || !is_array($_SESSION['suite_messages'][$type])
+            ) {
+                $_SESSION['suite_messages'][$type] = array();
+            }
+            if (($repeat)
+                || (!in_array($message, $_SESSION['suite_messages'][$type]))
+            ) {
+                $_SESSION['suite_messages'][$type][] = $message;
+            }
         }
     }
 
-    /**
-     * picking up the messages from the session and clearing session storage array
-     * @return array messages
+
+   /**
+     * Return all messages that have been set.
+     *
+     * @param string $type optional, values allowed: error, info, alert, okay, warning.
+     *
+     * @param boolean $clear_qeue optional, set to FALSE if you do not want to clear the messages queue
+     *
+     * @return associative array, the key is the message type, the value an array of messages.
+     * If the $type parameter is passed, you get only that type, or an empty array if there are no such messages.
+     * If $type is not passed, all message types are returned, or an empty array if none exist.
      */
-    protected static function getMessages($type) {
-        
-        self::validateMessageType($type);
+    public static function getMessages($type = null, $clear_queue = true)
+    {
+        $messages = array();
 
-        if (isset($_SESSION[$type]) && is_array($_SESSION[$type])) {
-            $msgs = $_SESSION[$type];
-            unset($_SESSION[$type]);
-            return $msgs;
-        } else {
-            return array();
+        if (!empty($_SESSION['user_error_message'])) {
+            SugarApplication::appendMessage($_SESSION['user_error_message'], 'error');
         }
-    }
 
-    /**
-     * 
-     * @param string $type possible message types: ['user_error_message', 'user_success_message']
-     * @throws Exception message type should be valid
-     */
-    protected static function validateMessageType($type) {
```

Comment:
Do not remove the validation of message type. (for e.g. nothing to shows up a misspelled index enimore?)

Reply:
You are right! Sorry!
Fixed in commit: https://github.com/salesagility/SuiteCRM/pull/6602/commits/7d5473201c4a1d912eb4378418ad8ca73bd3e344